File: web/src/api/tag.js
import request from '@/utils/request'

export function listTag (params) {
  return request({
    url: '/api/v1/tags',
    method: 'get',
    params
  })
}

export function createTag (data) {
  return request({
    url: '/api/v1/tag',
    method: 'post',
    data
  })
}

export function updateTag (data) {
  return request({
    url: '/api/v1/tag',
    method: 'put',
    data
  })
}

export function getTag (id) {
  return request({
    url: '/api/v1/tag',
    method: 'get',
    params: { id }
  })
}

export function deleteTag (id) {
  return request({
    url: '/api/v1/tag',
    method: 'delete',
    params: { id }
  })
}

File: server/internal/dao/tag.go
package dao

import (
	"blog-service/internal/app"
	"blog-service/internal/model"
)

func (d *Dao) GetTag(id uint32, state uint8) (model.Tag, error) {
	tag := model.Tag{ID: id, State: state}
	return tag.Get(d.engine)
}

func (d *Dao) ListTag(name string, state uint8, pageNum, pageSize int) ([]*model.Tag, error) {
	tag := model.Tag{Name: name, State: state}
	return tag.List(d.engine, app.GetPageOffset(pageNum, pageSize), pageSize)
}

func (d *Dao) CountTag(name string, state uint8) (int, error) {
	tag := model.Tag{Name: name, State: state}
	return tag.Count(d.engine)
}

func (d *Dao) CreateTag(name string, state uint8, createdBy string) error {
	tag := model.Tag{Name: name, CreatedBy: createdBy, State: state}
	_, err := tag.Create(d.engine)
	return err
}

func (d *Dao) UpdateTag(id uint32, name string, state uint8, modifiedBy string) error {
	tag := model.Tag{ID: id}
	values := map[string]interface{}{
		"state":       state,
		"modified_by": modifiedBy,
	}
	if name != "" {
		values["name"] = name
	}
	return tag.Update(d.engine, values)
}

func (d *Dao) DeleteTag(id uint32) error {
	tag := model.Tag{ID: id}
	return tag.Delete(d.engine)
}
